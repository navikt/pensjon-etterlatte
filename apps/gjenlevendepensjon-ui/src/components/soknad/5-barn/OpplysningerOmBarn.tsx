import React, { useState } from "react";
import ikon from "../../../assets/ikoner/barn1.svg";
import SoknadSteg from "../../../typer/SoknadSteg";
import { useSoknadContext } from "../../../context/soknad/SoknadContext";
import { IBarn, IOmBarn } from "../../../typer/person";
import { ActionTypes } from "../../../context/soknad/soknad";
import { useTranslation } from "react-i18next";
import BarnInfokort from "./BarnInfokort";
import LeggTilBarnSkjema from "./LeggTilBarnSkjema";
import { SkjemaGruppe } from "../../felles/SkjemaGruppe";
import { v4 as uuid } from "uuid";
import Navigasjon from "../../felles/Navigasjon";
import { Alert, BodyShort, Button, Modal, Panel, Heading } from "@navikt/ds-react";
import { FieldArrayWithId, FormProvider, useFieldArray, useForm } from "react-hook-form";
import { RHFSpoersmaalRadio } from "../../felles/rhf/RHFRadio";
import { deepCopy } from "../../../utils/deepCopy";
import AndreStoenader from "./AndreStoenader";
import styled from "styled-components";
import {Infokort, InfokortHeader, InfokortInformasjonsboks} from "../../felles/StyledComponents";
import {SkjemaElement} from "../../felles/SkjemaElement";

if (process.env.NODE_ENV !== "test") Modal.setAppElement!!("#root"); //Denne er ogsÃ¥ definert i Navigasjon. Trenger vi den?

const OpplysningerOmBarn: SoknadSteg = ({ neste, forrige }) => {
    const { t } = useTranslation();
    const { state, dispatch } = useSoknadContext();

    const methods = useForm<IOmBarn>({
        defaultValues: state.opplysningerOmBarn || {},
    });

    const {
        watch,
        getValues
    } = methods;

    const erValidert = state.opplysningerOmBarn.erValidert
    const registrerteBarn = watch("barn")

    const getFnrRegistrerteBarn = (): string[] => {
        if (registrerteBarn !== undefined) {
            return registrerteBarn.map(barn => barn.foedselsnummer !== undefined ? barn.foedselsnummer : "")
        } else {
            return []
        }
    }

    const fnrRegistrerteBarn = (aktivBarnIndex: number): string[] => {
        const fnr = getFnrRegistrerteBarn()
        fnr.splice(aktivBarnIndex, 1)
        return fnr
    }

    const { fields, append, update, remove } = useFieldArray({
        name: "barn",
        control: methods.control,
    });

    const [aktivBarnIndex, setAktivBarnIndex] = useState<number | undefined>(undefined)

    const leggtilNyttBarn = () => {
        append({})
        setAktivBarnIndex(fields.length)
    }

    const fjernNyttBarn = () => {
        remove(aktivBarnIndex)
    }

    const fjernBarn = (index: number) => {
        remove(index)
        const verdier = getValues()
        dispatch({ type: ActionTypes.OPPDATER_OM_BARN, payload: { ...deepCopy(verdier) } });
    }

    const oppdaterBarn = (barn: IBarn) => {
        if (aktivBarnIndex !== undefined) {
            update(aktivBarnIndex, barn)
            const verdier = getValues()
            dispatch({ type: ActionTypes.OPPDATER_OM_BARN, payload: { ...deepCopy(verdier) } });
        }
        setAktivBarnIndex(undefined)
    }

    const lagreNeste = (data: IOmBarn) => {
        dispatch({ type: ActionTypes.OPPDATER_OM_BARN, payload: { ...deepCopy(data), erValidert: true } });
        neste!!();
    };

    const lagreTilbake = (data: IOmBarn) => {
        dispatch({ type: ActionTypes.OPPDATER_OM_BARN, payload: { ...deepCopy(data), erValidert: true } })
        forrige!!()
    }

    const lagreTilbakeUtenValidering = () => {
        const verdier = getValues()
        dispatch({ type: ActionTypes.OPPDATER_OM_BARN, payload: { ...deepCopy(verdier), erValidert: false } })
        forrige!!()
    }

    const { handleSubmit } = methods;

    return (
        <FormProvider {...methods}>
            <form>
                {aktivBarnIndex === undefined &&
                <>
                    <SkjemaElement>
                        <Heading size={"medium"} className={"center"}>
                            {t("omBarn.tittel")}
                        </Heading>
                    </SkjemaElement>

                    <SkjemaElement>
                        <Panel border>
                            <Alert variant={"info"} className={"navds-alert--inline"}>
                                <BodyShort size={"small"}>{t("omBarn.informasjon")}</BodyShort>
                            </Alert>
                        </Panel>
                    </SkjemaElement>

                    <SkjemaGruppe>
                        <InfokortWrapper>
                            {fields?.map((field: FieldArrayWithId, index: number) => (
                                <BarnInfokort key={uuid()} barn={field as IBarn} index={index} fjern={fjernBarn}
                                              setAktivBarnIndex={() => setAktivBarnIndex(index)}/>
                            ))}

                            <Infokort>
                                <InfokortHeader gjennomsiktig>
                                    <img alt="barn" src={ikon}/>
                                </InfokortHeader>
                                <InfokortInformasjonsboks>
                                    <LeggTilBarnKnappWrapper>
                                        <Button data-testid={"legg-til-barn-knapp"} variant={"primary"} type={"button"} onClick={leggtilNyttBarn}>
                                            {t("knapp.leggTilBarn")}
                                        </Button>
                                    </LeggTilBarnKnappWrapper>

                                    <BodyShort size={"small"} className={"center mute"}>
                                        {t("omBarn.valgfritt")}
                                    </BodyShort>
                                </InfokortInformasjonsboks>
                            </Infokort>
                        </InfokortWrapper>
                    </SkjemaGruppe>
                    <SkjemaElement>
                        <RHFSpoersmaalRadio name={"gravidEllerNyligFoedt"} legend={t("omBarn.gravidEllerNyligFoedt")}/>
                    </SkjemaElement>

                    <AndreStoenader soeknad={state} barn={registrerteBarn} />

                    <Navigasjon
                        forrige={{ onClick: erValidert === true ? handleSubmit(lagreTilbake) : lagreTilbakeUtenValidering }}
                        neste={{ onClick: handleSubmit(lagreNeste) }}
                    />
                </>
                }

                {aktivBarnIndex !== undefined &&
                <LeggTilBarnSkjema lagre={oppdaterBarn} avbryt={() => setAktivBarnIndex(undefined)}
                                   fnrRegistrerteBarn={fnrRegistrerteBarn(aktivBarnIndex)}
                                   barn={fields[aktivBarnIndex] as IBarn}
                                   fjernAvbruttNyttBarn={fjernNyttBarn}
                />
                }

            </form>
        </FormProvider>
    );
};

export default OpplysningerOmBarn;

const InfokortWrapper = styled.div`
  display: flex;
  flex-flow: row wrap;
  margin: 0 auto;
  column-gap: 1rem;
`

const LeggTilBarnKnappWrapper = styled.div`
    text-align: center;
    margin-bottom: 1rem;
`