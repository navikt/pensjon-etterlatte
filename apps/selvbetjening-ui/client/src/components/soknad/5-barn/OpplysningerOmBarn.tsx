import React, { useState } from "react";
import "../../felles/Infokort.scss";
import ikon from "../../../assets/ikoner/barn1.svg";
import SoknadSteg from "../../../typer/SoknadSteg";
import { useSoknadContext } from "../../../context/soknad/SoknadContext";
import { IBarn, IOmBarn } from "../../../typer/person";
import { ActionTypes } from "../../../context/soknad/soknad";
import { useTranslation } from "react-i18next";
import BarnInfokort from "./BarnInfokort";
import LeggTilBarnSkjema from "./LeggTilBarnSkjema";
import { SkjemaGruppe } from "nav-frontend-skjema";
import { v4 as uuid } from "uuid";
import Navigasjon from "../../felles/Navigasjon";
import { Alert, BodyShort, Button, Modal, Panel, Heading } from "@navikt/ds-react";
import { FieldArrayWithId, FormProvider, useFieldArray, useForm } from "react-hook-form";
import { RHFSpoersmaalRadio } from "../../felles/RHFRadio";
import { deepCopy } from "../../../utils/deepCopy";
import AndreStoenader from "./AndreStoenader";

if (process.env.NODE_ENV !== "test") Modal.setAppElement!!("#root"); //Denne er ogsÃ¥ definert i Navigasjon. Trenger vi den?

const OpplysningerOmBarn: SoknadSteg = ({ neste, forrige }) => {
    const { t } = useTranslation();
    const { state, dispatch } = useSoknadContext();

    const methods = useForm<IOmBarn>({
        defaultValues: state.opplysningerOmBarn || {},
    });

    const {
        watch,
        getValues
    } = methods;

    const erValidert = state.opplysningerOmBarn.erValidert
    const registrerteBarn = watch("barn")

    const getFnrRegistrerteBarn = (): string[] => {
        if (registrerteBarn !== undefined) {
            return registrerteBarn.map(barn => barn.foedselsnummer !== undefined ? barn.foedselsnummer : "")
        } else {
            return []
        }
    }

    const fnrRegistrerteBarn = (aktivBarnIndex: number): string[] => {
        const fnr = getFnrRegistrerteBarn()
        fnr.splice(aktivBarnIndex, 1)
        return fnr
    }

    const { fields, append, update, remove } = useFieldArray({
        name: "barn",
        control: methods.control,
    });

    const [aktivBarnIndex, setAktivBarnIndex] = useState<number | undefined>(undefined)

    const leggtilNyttBarn = () => {
        append({})
        setAktivBarnIndex(fields.length)
    }

    const fjernNyttBarn = () => {
        remove(aktivBarnIndex)
    }

    const oppdaterBarn = (barn: IBarn) => {
        if (aktivBarnIndex !== undefined) {
            update(aktivBarnIndex, barn)
        }
        setAktivBarnIndex(undefined)
    }

    const lagreNeste = (data: IOmBarn) => {
        dispatch({ type: ActionTypes.OPPDATER_OM_BARN, payload: { ...deepCopy(data), erValidert: true } });
        neste!!();
    };

    const lagreTilbake = (data: IOmBarn) => {
        dispatch({ type: ActionTypes.OPPDATER_OM_BARN, payload: { ...deepCopy(data), erValidert: true } })
        forrige!!()
    }

    const lagreTilbakeUtenValidering = () => {
        const verdier = getValues()
        dispatch({ type: ActionTypes.OPPDATER_OM_BARN, payload: { ...deepCopy(verdier), erValidert: false } })
        forrige!!()
    }

    const { handleSubmit } = methods;

    return (
        <FormProvider {...methods}>
            <form>
                {aktivBarnIndex === undefined &&
                <>
                    <SkjemaGruppe>
                        <Heading size={"medium"} className={"center"}>
                            {t("omBarn.tittel")}
                        </Heading>
                    </SkjemaGruppe>

                    <SkjemaGruppe>
                        <Panel border>
                            <Alert variant={"info"} className={"navds-alert--inline"}>
                                <BodyShort size={"small"}>{t("omBarn.informasjon")}</BodyShort>
                            </Alert>
                        </Panel>
                    </SkjemaGruppe>

                    <SkjemaGruppe>
                        <div className={"infokort-wrapper"}>
                            {fields?.map((field: FieldArrayWithId, index: number) => (
                                <BarnInfokort key={uuid()} barn={field as IBarn} index={index} fjern={remove}
                                              setAktivBarnIndex={() => setAktivBarnIndex(index)}/>
                            ))}

                            <div className={"infokort"}>
                                <div className={"infokort__header gjennomsiktig"}>
                                    <img alt="barn" className="barneikon" src={ikon}/>
                                </div>
                                <div className={"infokort__informasjonsboks"}>
                                    <div className={"informasjonsboks-innhold"}>
                                        <Button variant={"primary"} type={"button"} onClick={leggtilNyttBarn}>
                                            {t("knapp.leggTilBarn")}
                                        </Button>
                                    </div>

                                    <BodyShort size={"small"} className={"center mute"}>
                                        {t("omBarn.valgfritt")}
                                    </BodyShort>
                                </div>
                            </div>
                        </div>
                    </SkjemaGruppe>
                    <RHFSpoersmaalRadio name={"gravidEllerNyligFoedt"} legend={t("omBarn.gravidEllerNyligFoedt")}/>

                    <AndreStoenader soeknad={state} barn={registrerteBarn} />

                    <Navigasjon
                        forrige={{ onClick: erValidert === true ? handleSubmit(lagreTilbake) : lagreTilbakeUtenValidering }}
                        neste={{ onClick: handleSubmit(lagreNeste) }}
                    />
                </>
                }

                {aktivBarnIndex !== undefined &&
                <LeggTilBarnSkjema lagre={oppdaterBarn} avbryt={() => setAktivBarnIndex(undefined)}
                                   fnrRegistrerteBarn={fnrRegistrerteBarn(aktivBarnIndex)}
                                   barn={fields[aktivBarnIndex] as IBarn}
                                   fjernAvbruttNyttBarn={fjernNyttBarn}
                />
                }

            </form>
        </FormProvider>
    );
};

export default OpplysningerOmBarn;
