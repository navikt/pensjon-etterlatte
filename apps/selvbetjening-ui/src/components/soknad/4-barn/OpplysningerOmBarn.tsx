import React, { useState } from "react";
import "../../../App.less";
import "../../felles/Infokort.less";
import ikon from "../../../assets/ikoner/barn1.svg";
import { Systemtittel } from "nav-frontend-typografi";
import { Hovedknapp, Knapp } from "nav-frontend-knapper";
import SoknadSteg from "../../../typer/SoknadSteg";
import { default as Modal } from "nav-frontend-modal";
import { useSoknadContext } from "../../../context/soknad/SoknadContext";
import { IBarn } from "../../../typer/person";
import { ActionTypes } from "../../../context/soknad/soknad";
import { useTranslation } from "react-i18next";
import BarnInfokort from "./BarnInfokort";
import LeggTilBarnSkjema from "./LeggTilBarnSkjema";
import { BekreftCheckboksPanel, SkjemaGruppe } from "nav-frontend-skjema";
import { v4 as uuid } from "uuid";

Modal.setAppElement("#root");

const OpplysningerOmBarn: SoknadSteg = ({ neste, forrige }) => {
    const { t } = useTranslation();
    const { state, dispatch } = useSoknadContext();

    // Modal
    const [isOpen, setIsOpen] = useState(false);

    const leggTilBarn = (data: IBarn) => {
        dispatch({ type: ActionTypes.LEGG_TIL_BARN, payload: data });
        setIsOpen(false);
    };

    const fjern = (index: number) =>
        dispatch({ type: ActionTypes.FJERN_BARN, payload: index, });

    const [soekerBarnep, setSoekerBarnep] = useState<boolean>(false);

    return (
        <>
            {/* Steg 4 */}
            <SkjemaGruppe>
                <Systemtittel className={"center"}>
                    Om barn
                </Systemtittel>
            </SkjemaGruppe>

            <SkjemaGruppe>
                <div className={"infokort-wrapper"}>
                    {state.opplysningerOmBarn?.map((barn: IBarn, index: number) => (
                        <BarnInfokort key={uuid()} barn={barn} index={index} fjern={fjern} />
                    ))}

                    <div className={"infokort"}>
                        <div className={"infokort__header gjennomsiktig"}>
                            <img alt="barn" className="barneikon" src={ikon} />
                        </div>
                        <div className={"infokort__informasjonsboks"}>
                            <div className={"informasjonsboks-innhold"}>
                                <Knapp onClick={() => setIsOpen(true)}>
                                    {t("knapp.leggTilBarn")}
                                </Knapp>
                            </div>
                        </div>
                    </div>
                </div>

                <Modal
                    isOpen={isOpen}
                    onRequestClose={() => setIsOpen(false)}
                    closeButton={true}
                    contentLabel={t("omBarn.tittel")}
                >
                    <LeggTilBarnSkjema lagre={leggTilBarn} />
                </Modal>
            </SkjemaGruppe>

            {state.opplysningerOmBarn.length > 0 && (
                <SkjemaGruppe>
                    <BekreftCheckboksPanel
                        label={"Ja, jeg ønsker å søke om barnepensjon."}
                        checked={soekerBarnep}
                        onChange={(e) => setSoekerBarnep((e.target as HTMLInputElement).checked)}
                    >
                        Hvis du har felles barn med avdøde, som er under 18 år, kan de ha rett til barnepensjon.
                    </BekreftCheckboksPanel>
                </SkjemaGruppe>
            )}

            <SkjemaGruppe className={"navigasjon-rad"}>
                <Knapp htmlType={"button"} onClick={forrige}>{t("knapp.tilbake")}</Knapp>
                <Hovedknapp onClick={neste}>{t("knapp.neste")}</Hovedknapp>
            </SkjemaGruppe>
        </>
    );
};

export default OpplysningerOmBarn;
