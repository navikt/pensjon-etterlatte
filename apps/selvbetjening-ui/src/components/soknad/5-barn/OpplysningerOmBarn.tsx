import React, { useState } from "react";
import "../../felles/Infokort.less";
import ikon from "../../../assets/ikoner/barn1.svg";
import { Normaltekst, Systemtittel } from "nav-frontend-typografi";
import { Knapp } from "nav-frontend-knapper";
import SoknadSteg from "../../../typer/SoknadSteg";
import { default as Modal } from "nav-frontend-modal";
import { useSoknadContext } from "../../../context/soknad/SoknadContext";
import { GravidEllerNyligFoedt, IBarn, IOmBarn } from "../../../typer/person";
import { ActionTypes } from "../../../context/soknad/soknad";
import { useTranslation } from "react-i18next";
import BarnInfokort from "./BarnInfokort";
import LeggTilBarnSkjema from "./LeggTilBarnSkjema";
import { RadioProps, SkjemaGruppe } from "nav-frontend-skjema";
import { v4 as uuid } from "uuid";
import Navigasjon from "../../felles/Navigasjon";
import AlertStripe from "nav-frontend-alertstriper";
import { FieldArrayWithId, FormProvider, useFieldArray, useForm } from "react-hook-form";
import { RHFRadio } from "../../felles/RHFRadio";
import Panel from "nav-frontend-paneler";
import { useEffectOnce } from "../../../utils/extensions";
import { isEmpty } from "lodash";

Modal.setAppElement("#root");

const OpplysningerOmBarn: SoknadSteg = ({ neste, forrige }) => {
    const { t } = useTranslation();
    const { state, dispatch } = useSoknadContext();

    const methods = useForm<IOmBarn>({
        defaultValues: state.opplysningerOmBarn || {}
    });

    useEffectOnce(() => {
        methods.reset(state.opplysningerOmBarn)
    }, !isEmpty(state.opplysningerOmBarn));

    const { fields, append, remove } = useFieldArray({
        name: "barn",
        control: methods.control
    });

    // Modal
    const [isOpen, setIsOpen] = useState(false);

    const leggTilBarn = (data: IBarn) => {
        append(data as any)
        setIsOpen(false);
    };

    const lagre = (data: IOmBarn) => {
        dispatch({ type: ActionTypes.OPPDATER_OM_BARN, payload: data });
        neste!!();
    }

    const { handleSubmit } = methods;

    return (
        <FormProvider {...methods}>
            <form>
                <SkjemaGruppe>
                    <Systemtittel className={"center"}>
                        {t("omBarn.tittel")}
                    </Systemtittel>
                </SkjemaGruppe>

                <SkjemaGruppe>
                    <Panel border>
                        <AlertStripe type={"info"} form={"inline"}>
                            <Normaltekst>
                                {t("omBarn.informasjon")}
                            </Normaltekst>
                        </AlertStripe>
                    </Panel>
                </SkjemaGruppe>

                <SkjemaGruppe>
                    <div className={"infokort-wrapper"}>
                        {fields?.map((field: FieldArrayWithId, index: number) => (
                            <BarnInfokort
                                key={uuid()}
                                barn={field as IBarn}
                                index={index}
                                fjern={remove}
                            />
                        ))}

                        <div className={"infokort"}>
                            <div className={"infokort__header gjennomsiktig"}>
                                <img alt="barn" className="barneikon" src={ikon} />
                            </div>
                            <div className={"infokort__informasjonsboks"}>
                                <div className={"informasjonsboks-innhold"}>
                                    <Knapp htmlType={"button"} onClick={() => setIsOpen(true)}>
                                        {t("knapp.leggTilBarn")}
                                    </Knapp>
                                </div>

                                <Normaltekst className={"center mute"}>
                                    {t("omBarn.valgfritt")}
                                </Normaltekst>
                            </div>
                        </div>
                    </div>

                    <Modal
                        isOpen={isOpen}
                        onRequestClose={() => setIsOpen(false)}
                        closeButton={true}
                        contentLabel={t("omBarn.tittel")}
                    >
                        <LeggTilBarnSkjema lagre={leggTilBarn} />
                    </Modal>
                </SkjemaGruppe>

                <RHFRadio
                    name={"gravidEllerNyligFoedt"}
                    legend={t("omBarn.gravidEllerNyligFoedt")}
                    radios={Object.values(GravidEllerNyligFoedt).map(value => {
                        return { label: t(value), value } as RadioProps
                    })}
                />

                <Navigasjon
                    forrige={{ onClick: forrige }}
                    neste={{ onClick: handleSubmit(lagre) }}
                />
            </form>
        </FormProvider>
    );
};

export default OpplysningerOmBarn;
