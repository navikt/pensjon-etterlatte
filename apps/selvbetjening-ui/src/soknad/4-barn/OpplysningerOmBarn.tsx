import React, { FC } from "react";
import "../../App.less";
import { Panel } from "nav-frontend-paneler";
import { Input, Radio, RadioGruppe, SkjemaGruppe } from "nav-frontend-skjema";
import { Systemtittel } from "nav-frontend-typografi";
import { Hovedknapp, Knapp } from "nav-frontend-knapper";
import { useHistory } from "react-router-dom";

interface Props {
    forrigeSteg?: number;
    nesteSteg?: number;
}

const OpplysningerOmBarn: FC<Props> = ({ forrigeSteg, nesteSteg }) => {
    const history = useHistory();

    return (
        <div className="app">
            {/* Steg 4 */}
            <Panel>
                <Systemtittel>4 Opplysninger om barn</Systemtittel>

                <SkjemaGruppe>
                    <Input label="Fornavn" />
                    <Input label="Etternavn" />
                    <Input label="Fødselsnummer (11 siffer)" />

                    <RadioGruppe>
                        <Radio label={"Fellesbarn m/avdøde"} name="" />
                        <Radio label={"Avdødes særkullsbarn"} name="" />
                        <Radio label={"Egne særkullsbarn"} name="" />
                    </RadioGruppe>

                    <RadioGruppe legend="Bor barnet i utlandet?">
                        <Radio label={"Ja"} name="" />
                        <Radio label={"Nei"} name="" />
                    </RadioGruppe>

                    <Input label=" Hvis ja, oppgi statsborgerskap og land." />
                </SkjemaGruppe>

                <br />
                <Knapp>+ Legg til barn</Knapp>
            </Panel>

            <Panel>
                {forrigeSteg && <Knapp onClick={() => history.push(`/soknad/steg/${forrigeSteg}`)}>Tilbake</Knapp>}
                {nesteSteg && <Hovedknapp onClick={() => history.push(`/soknad/steg/${nesteSteg}`)}>Neste</Hovedknapp>}
            </Panel>
        </div>
    );
};

export default OpplysningerOmBarn;
