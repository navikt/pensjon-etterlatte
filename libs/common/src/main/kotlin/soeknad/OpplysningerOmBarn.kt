package soeknad

import com.fasterxml.jackson.annotation.JsonValue
import no.nav.etterlatte.libs.common.soeknad.Valg


data class OpplysningerOmBarn(
    val barn: List<Barn> = emptyList(),
    val gravidEllerNyligFoedt: Valg
)

data class Barn(
    val fornavn: String,
    val etternavn: String,
    val foedselsnummer: String,
    val harBarnetVerge: HarBarnetVerge?,
    val relasjon: BarnRelasjonType?,
    val statsborgerskap: String?,
    val bosattUtland: BosattUtland?,
    val dagligOmsorg: Valg?,
    val barnepensjon: Barnepensjon?
)

data class HarBarnetVerge(
    val svar: Valg,
    val fornavn: String?,
    val etternavn: String?,
    val foedselsnummer: String?
)

enum class BarnRelasjonType(@get:JsonValue val beskrivelse: String) {
    fellesbarnMedAvdoede("barnRelasjon.fellesbarnMedAvdoede"),
    avdoedesSaerkullsbarn("barnRelasjon.avdoedesSaerkullsbarn"),
    egneSaerkullsbarn("barnRelasjon.egneSaerkullsbarn"),
}

data class BosattUtland(
    val svar: Valg,
    val land: String?,
    val adresse: String?
)

data class Barnepensjon(
    val soeker: Valg?,
    val kontonummer: Kontonummer?,
    val forskuddstrekk: Forskuddstrekk?,
)

data class Kontonummer(
    val svar: Valg,
    val kontonummer: String?
)

data class Forskuddstrekk(
    val svar: Valg,
    val trekkprosent: String?
)
