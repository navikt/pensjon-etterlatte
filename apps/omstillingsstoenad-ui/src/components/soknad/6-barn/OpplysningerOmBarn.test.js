import { render } from '@testing-library/react'
import OpplysningerOmBarn from './OpplysningerOmBarn'

jest.mock('react-i18next', () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    ...jest.requireActual('react-i18next'),
    useTranslation: () => {
        return {
            t: (str) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        }
    },
}))

describe('Om den avdÃ¸de', () => {
    it('Snapshot', () => {
        const { container } = render(<OpplysningerOmBarn />)
        expect(container).toMatchSnapshot()
    })
})
