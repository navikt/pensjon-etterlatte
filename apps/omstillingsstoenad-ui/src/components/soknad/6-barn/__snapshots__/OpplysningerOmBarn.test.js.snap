// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Om den avd√∏de Snapshot 1`] = `
<div>
  <form>
    <div
      class="sc-jsMahE iLNtlK"
    >
      <h1
        class="center navds-heading navds-heading--medium"
      >
        omBarn.tittel
      </h1>
    </div>
    <div
      class="sc-jsMahE iLNtlK"
    >
      <div
        class="navds-panel navds-panel--border"
      >
        <div
          class="navds-alert--inline navds-alert navds-alert--info navds-alert--medium"
        >
          <svg
            aria-labelledby="title-r0"
            class="navds-alert__icon"
            fill="none"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="title-r0"
            >
              Informasjon
            </title>
            <path
              clip-rule="evenodd"
              d="M12 0c6.627 0 12 5.373 12 12s-5.373 12-12 12S0 18.627 0 12 5.373 0 12 0ZM9 19v-2h2v-5H9v-2h4v7h2v2H9Zm3-14a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <div
            class="navds-alert__wrapper navds-body-long"
          >
            <p
              class="navds-body-short navds-body-short--small"
            >
              omBarn.informasjon
            </p>
          </div>
        </div>
      </div>
    </div>
    <fieldset
      class="sc-beqWaB kWPIyU"
    >
      <div
        class="sc-hIqOWS jDiEwI"
      >
        <div
          class="sc-gLDzan jZdOjA"
        >
          <div
            class="sc-eDDNvR gNRAeQ"
          >
            <img
              alt="barn"
              src="barn1.svg"
            />
          </div>
          <div
            class="sc-jTrPJq dWNTiq"
          >
            <div
              class="sc-dnwKUv cOiLvv"
            >
              <button
                class="navds-button navds-button--primary navds-button--medium"
                data-testid="legg-til-barn-knapp"
                type="button"
              >
                <span
                  aria-live="polite"
                  class="navds-label"
                >
                  knapp.leggTilBarn
                </span>
              </button>
            </div>
            <p
              class="center mute navds-body-short navds-body-short--small"
            >
              omBarn.valgfritt
            </p>
          </div>
        </div>
      </div>
    </fieldset>
    <div
      class="sc-jsMahE iLNtlK"
    >
      <div
        id="gravidEllerNyligFoedt"
      >
        <fieldset
          aria-invalid="false"
          class="sc-fbJfA jbbuKd navds-radio-group navds-radio-group--medium navds-fieldset navds-fieldset--medium"
        >
          <legend
            class="navds-fieldset__legend navds-label"
          >
            omBarn.gravidEllerNyligFoedt
          </legend>
          <div
            class="navds-radio-buttons"
          >
            <div
              class="radioBorder navds-radio navds-radio--medium"
            >
              <input
                aria-invalid="false"
                class="navds-radio__input"
                id="radio-r3"
                name="gravidEllerNyligFoedt"
                type="radio"
                value="radiobuttons.ja"
              />
              <label
                class="navds-radio__label"
                for="radio-r3"
              >
                <span
                  class="navds-radio__content"
                >
                  <span
                    class="navds-body-short"
                  >
                    radiobuttons.ja
                  </span>
                </span>
              </label>
            </div>
            <div
              class="radioBorder navds-radio navds-radio--medium"
            >
              <input
                aria-invalid="false"
                class="navds-radio__input"
                id="radio-r4"
                name="gravidEllerNyligFoedt"
                type="radio"
                value="radiobuttons.nei"
              />
              <label
                class="navds-radio__label"
                for="radio-r4"
              >
                <span
                  class="navds-radio__content"
                >
                  <span
                    class="navds-body-short"
                  >
                    radiobuttons.nei
                  </span>
                </span>
              </label>
            </div>
          </div>
          <div
            aria-live="polite"
            aria-relevant="additions removals"
            class="navds-fieldset__error"
            id="fieldset-error-r2"
          />
        </fieldset>
      </div>
    </div>
    <fieldset
      class="sc-beqWaB kWPIyU"
    >
      <div
        class="sc-gueYoa FJQFS undefined"
      >
        <button
          class="navds-button navds-button--primary navds-button--medium"
          type="button"
        >
          <span
            aria-live="polite"
            class="navds-label"
          >
            knapp.tilbake
          </span>
        </button>
        <button
          class="navds-button navds-button--primary navds-button--medium"
          type="button"
        >
          <span
            aria-live="polite"
            class="navds-label"
          >
            knapp.neste
          </span>
        </button>
      </div>
    </fieldset>
    <div
      class="sc-gueYoa sc-dmqHEX FJQFS fYyOWz"
    >
      <button
        class="navds-button navds-button--secondary navds-button--medium"
        id="avbryt-btn"
        type="button"
      >
        <span
          aria-live="polite"
          class="navds-label"
        >
          knapp.avbryt
        </span>
      </button>
    </div>
  </form>
</div>
`;
